### PR TITLE
refactor: [M3-6437] - MUI v5 Migration - Update src/components grid references

### PR DESCRIPTION
**Pull request by jaalah-akamai**
## Description 📝
Migrating to MUI Grid v2. This covers everything in the `src/components` directory and any additional files that needed to be cleaned up.

## Major Changes 🔄 
- Changed `Grid` import from `src/components/Grid` to `@mui/material/Unstable_Grid2`. 
   - The `item` attribute is no longer needed for `Grid`.
   - Most `Grid` elements with the `container` attribute needed a `spacing` attribute as part of the change
- `ImageSelect/ImageSelect.tsx`: Removed extra unnecessary `Grid` wrappers.
- `PaymentDrawer/CreditCard.tsx`: Updated JSX to match `ThirdPartyPayment.tsx` structure to get things in sync. We can look at maybe re-using some things between the two.

### Replaced Grid w/ Box component
There was no need for a `Grid` components in these files, so I swapped them with `Box`
- `StackScript/StackScript.tsx`
- `PaymentMethodRow/ThirdPartyPayment.tsx`
- `PaymentMethodRow/PaymentMethodRow.tsx`: Completely refactored styles
- `ImageSelect/ImageOption.tsx`
- `TagCell/TagCell.tsx`

### Deleted
- `components/LineGraph/SimpleLegend.tsx`: Not used anymore. Removed all related files/references.
- `components/Skeleton/Skeleton.tsx`: Not used anymore. Removed all related files/references. We import `Skeleton` directly from library in other places.
- `TableHeader/TableHeader.tsx`: Not used anymore. Removed all related files/references.

## How to test
| Component    | Where to view? | Screenshot |
| ----------- | ----------- | ----------- |
| AbuseTicketBanner.tsx<br /><br />http://localhost:3000/ | Turn on MSW | abusebanner|
| Accordion.tsx<br /><br /> http://localhost:3000/linodes/{ID}/settings | Linode detail "Settings" tab or Storybook |  |
| Dialog.tsx<br />http://localhost:3000/linodes |   Delete linode dialog | dialog|
| DismissibleBanner.tsx<br />http://localhost:3000/ |   Turn on MSW | dismissable|
| Drawer.tsx<br />http://localhost:3000/account/billing |   Any drawer here | |
| EditableEntityLabel.tsx<br />http://localhost:3000/longview |   Clients tab > longview editable label | longview|
| RegionOption.tsx<br />http://localhost:3000/linodes/create |   Region select dropdown | region|
| EntityDetail.tsx<br />http://localhost:3000/linodes/{ID} |   Linodes detail page, main summary card | entitydetail|
| ErrorState.tsx<br />http://localhost:3000/kubernetes/clusters |   Toggle error flag in code | error |
| ImageOption.tsx<br />http://localhost:3000/linodes/create |   Images select dropdown | images|
| MultipleIPInput.tsx<br />http://localhost:3000/domains/{ID} |   Edit any SOA Record domain | multipleip|
| PaginationFooter.tsx<br />http://localhost:3000/linodes |   Turn on MSW |pagination |
| PasswordInput.tsx & StrengthIndicator.tsx<br />http://localhost:3000/linodes/create |   | password|
| PaymentMethodRow.tsx & PaymentMethodRow/ThirdPartyPayment.tsx<br />http://localhost:3000/account/billing |   Turn on MSW | paymentmethod|
| StackScript.tsx<br />http://localhost:3000/stackscripts/{ID} |  Header is a Grid |stackscript |
| TabbedPanel.tsx<br />http://localhost:3000/linodes/create |   Linode Plan |  tabbed|
| TagCell.tsx<br />http://localhost:3000/linodes?view=grid |   "Add a tag" | tagcell|
| TransferDisplay.tsx<br />http://localhost:3000/linodes |   "Monthly Network Transfer Pool" in footer | transferdisplay|

### CODE REVIEW

File: packages/manager/src/components/Dialog/Dialog.tsx [hunk 1]
```diff
@@ -153,7 +142,7 @@ const Dialog: React.FC<DialogProps> = (props) => {
         </div>
         {titleBottomBorder && <hr className={classes.titleBottomBorder} />}
         <Grid container>
-          <div className={classNames(classes.dialogContent, className)}>
+          <div className={className}>
```
Comment (bnussman-akamai) [inline packages/manager/src/components/Dialog/Dialog.tsx:145]: Somewhat unrelated to your changes, but this component is very deceiving in how it applies the `className` to this body div and not the root Dialog. Generally speaking, are we ok with this?

File: packages/manager/src/components/PaginationFooter/PaginationFooter.tsx [hunk 2]
```diff
@@ -120,7 +120,7 @@ class PaginationFooter extends React.PureComponent<CombinedProps> {
           )}
         </Grid>
         {!fixedSize ? (
-          <Grid item className={`${classes.select} p0`}>
+          <Grid className={`${classes.select}`}>
```
Comment (bnussman-akamai) [inline packages/manager/src/components/PaginationFooter/PaginationFooter.tsx:123]:
```suggestion
          <Grid className={classes.select}>
```

**Review by bnussman-akamai — APPROVED** [review]
I didn't find any egregious differences from `develop` 🎉

File: packages/manager/src/components/Dialog/Dialog.tsx (hunk 1 above)
Comment (jaalah-akamai) [reply packages/manager/src/components/Dialog/Dialog.tsx:145]: Not ok with this, but I will fix it as part of the general refactor of that component when I get around to it 👍

**Review by mjac0bs — COMMENTED** [review]
This is looking good to me; thanks for the very thorough listing of components to check. I noticed one minor shift that I think we should fix in the StackScripts header, which is no longer left-aligned.

Can we get the merge conflicts resolved?

Prod:
Screenshot 2023-04-06 at 10 52 14 AM

This branch:
Screenshot 2023-04-06 at 10 52 03 AM

**Review by hana-linode — APPROVED** [review]
Nice cleanup! 🧹 🚀

**Pull request merged**